src/search.py

```diff
@@ -1,42 +1,61 @@
-import numpy as np
-from matplotlib.axes import Axes
-from statistics import mean
+from re import T
 from users import User
+from beam import Beam
 from typing import Union, Sequence
-from plotting import plot_beams
+from plotting import Cell
 
-def send_beam(start_angle, end_angle, users, cell_radius):
+def send_beam(beam: Beam, users, cell_radius):
     ack = 0
     for user in users:
-        if (user.theta >= start_angle) and (user.theta <= end_angle) and (user.radius <= cell_radius):
-            ack = ack + 1
+        if (user.theta >= beam.start_angle) and (user.theta <= beam.end_angle) and (user.radius <= cell_radius):
+            ack = 1
     return ack
 
-def round_robin_search(beams: np.ndarray, cell_radius, users: Union[User, Sequence[User]], tolerance, ax: Axes):
-    beam_list = []
-    marked_angles = []
-    # Search through beams once and mark beams with acknowledgements
-    for i in range(0, beams.size - 1):
-        acks = send_beam(beams[i], beams[i+1], users, cell_radius)
+def tolerance_met(beam_list: Union[Beam, Sequence[Beam]], tolerance):
+    flag = 1
+    for beam in beam_list:
+        if beam.span < tolerance:
+            flag = 0
+    return flag
```

Comment:
This should return True or False instead of 0 or 1.

Reply:
I agree